src/components/Courses/Courses.tsx

```diff
@@ -0,0 +1,48 @@
+import CourseCard from './component/CourseCard/CourseCard';
+import SearchBar from './component/SearchBar/SearchBar';
+
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import {
+	selectFilterField,
+	selectCoursesList,
+} from '../../store/courses/selectors';
+
+import { CoursesContainer } from './Courses.styles';
+
+const Courses = () => {
+	const filterField = useSelector(selectFilterField);
+	const coursesList = useSelector(selectCoursesList);
+
+	const [filderedCoursesList, setFilderedCoursesList] = useState(coursesList);
```

Comment:
No type for this `useState`

Reply:
![image](https://user-images.githubusercontent.com/90048259/228787643-0ef04516-8597-4b8f-abeb-ece0e5089446.png)
Why do you need to explicitly specify the TYPE here?
typescript gets it without any problems.
The key is that the type is already in the root-state

Reply:
I checked my version of TypeScript
And I have "typescript": "^4.5.4"
and it still infers everything correctly.
I don't mind setting the types, but it's doing some weird stuff, 
I just copy the correct and ready types from the VSCode prompts into my code...
![image](https://user-images.githubusercontent.com/90048259/228883101-5498cafe-f107-4225-8b19-a246e7757cf2.png)


![image](https://user-images.githubusercontent.com/90048259/228881966-4348a073-012c-443f-b9e5-ae3db17cf1ed.png)
![image](https://user-images.githubusercontent.com/90048259/228882277-e12c855c-901a-4f31-ba59-b09500d0d19f.png)
![image](https://user-images.githubusercontent.com/90048259/228882664-70f653b8-ba27-4560-879b-e50e75735486.png)